packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -572,6 +572,10 @@ function commitPassiveEffects(root: FiberRoot, firstEffect: Fiber): void {
   if (rootExpirationTime !== NoWork) {
     requestWork(root, rootExpirationTime);
   }
+  // Flush any sync work that was scheduled by effects
+  if (!isRendering) {
```

Comment:
I _think_ this is supposed to be `if (!isBatchingUpdates && !isRendering)` but, this seems generally correct

Reply:
Can you think of a case where this matters? I thought we always batched updates during passive effects.

Reply:
I would guess that this demonstrates it, but I didn't verify:

```
let _set;
function Foo() {
  _set = useState(0)[1];
  useEffect(() => {
    ReactDOM.render(update1, otherContainer);
  });
  return null;
}

render(<Foo />);
ReactDOM.unstable_batchedUpdates(() => {
  _set(0);  // forces effects to be flushed
  ReactDOM.render(update2, otherContainer);
});
```